### DISCUSSION TITLE
The unholy trio of stabilitydays, automerge, internalChecksFilter vs. PRs

### DISCUSSION THREAD

**Discussion by thoastbrot** [renovatebot/renovate#17569] [Request Help] [answered]
Hi,

I'm probably having an issue understanding the docs, as I'm expecting...

given the following configuration:

stabilitydays = 3
automerge = true
internalChecksFilter = strict

to unfold the following scenarios as described:

A: new version of a dependency is released in an existing repository

1. no branch creation (security concerns)
2. no PR creation

B: I'm running the bot on a new project with old deps to be updated (difference to A: the dependency fulfills the stabilitydays criteria)

1. branch is created
2. PR is created (and merged when the pipeline succeeds)

(without delay, both)

I cannot test A easily without creating a fake package, but as it's unverified, let's leave it at that for now.
For B, I have observed 1. but without a PR. It's created in the next run, I assume it's testing for the added stabilitydays-check job.

Why is that so? Without stability days, the automerge flow is doing fine for me - build is passing CI and it's done. I just would like to prepend a delay so upstream can react to issues like uaparser and the likes.

platform: Gitlab 15.2
package type: npm

**Comment by PhilipAbed** [comment]
I will need more info to investigate this:
1) Auto merge is dependent on the `platform` and on the `package manager`/`package type` and i'm not sure which ones you are using
2) i would like to see logs for the cases you are mentioning, it might be helpful, you can redact any sensitive information.
3) if you can create a Minimal Reproduction Repository it would be helpful also and the issue will be resolved faster.

**Comment by thoastbrot** [comment, marked as answer]
I'm sorry. "prCreation": "not-pending" was added as well when configuring the stabilityDays and I was too busy reading up on the effects of the props in the title.

Solution: read carefully what's configured.
